### DISCUSSION TITLE
File upload not working in GUI

### DISCUSSION THREAD

**Discussion by Vlad-Kliucharov** [ludeknovy/jtl-reporter#96] [Q&A] [answered]
File upload not working in GUI

https://user-images.githubusercontent.com/20796574/140309187-b687abda-1c67-444f-a1c2-088a9796c0a5.mp4

**Comment by ludeknovy** [comment]
@Vlad-Kliucharov Could you please try to upload only .jtl file?

And if you could provide me with the error logs upon uploading by running: `docker-compose logs be`

**Reply by ludeknovy** [reply]
@Vlad-Kliucharov did you make it work?

**Reply by Vlad-Kliucharov** [reply]
Hi @ludeknovy!
I try to upload only the .jtl file and the issue is reproduced

https://user-images.githubusercontent.com/20796574/140701958-82ef13f2-f89e-4520-b5c0-71aceece3b51.mp4

**Reply by Vlad-Kliucharov** [reply]
I run this command "docker-compose logs backend" where I can find the log file?

**Reply by ludeknovy** [reply]
You get the logs in the terminal. 

Would it be possible to share the kpi.jtl file (without any confidential information) here?

Btw. on the second video, all I can see is that the file was successfully uploaded - it takes some time to process it.

**Reply by Vlad-Kliucharov** [reply]
When I run this command nothing appear in the console

![Screenshot_2](https://user-images.githubusercontent.com/20796574/140702938-13c2ec58-0a53-42c4-9519-ba8f053faa3e.png)

**Reply by Vlad-Kliucharov** [reply]
[kpi.zip](https://github.com/ludeknovy/jtl-reporter/files/7495149/kpi.zip)

**Reply by ludeknovy** [reply]
@Vlad-Kliucharov I think I know what the issue is.

Not sure if I will be able to fix it for v3, but try it with the next major release and it worked for me just fine.
If you want to try that out, you just need to use a different docker compose file: https://github.com/ludeknovy/jtl-reporter/blob/feature/v4/docker-compose.yml

**Reply by ludeknovy** [reply]
I just pushed the fix for v3. Use the latest tag for backed service in docker-compose file: novyl/jtl-reporter-be

**Reply by Vlad-Kliucharov** [reply]
I had replaced a "docker-compose.yml", rebuild project and now on "Login" I have a blank screen
![Screenshot_3](https://user-images.githubusercontent.com/20796574/140898505-e423133b-1029-4331-923f-c611e288e939.png)

**Reply by ludeknovy** [reply]
@Vlad-Kliucharov you need to provide me more info here. Are there any errors in the console?
And what version did you use? v3 or v4?

I tested it yesterday in digital ocean VM and it worked fine for me.

**Reply by Vlad-Kliucharov** [reply]
Yes, there is an error in the console, and I had updated yml from the link which you pin in the comments
![Screenshot_3](https://user-images.githubusercontent.com/20796574/140912826-d5413d43-88ef-4c88-9ae4-e913a1129790.png)

**Reply by ludeknovy** [reply]
Ahh, my bad, sorry. I did not realize you need to also run a different DB. 

So if you want to try out the v4: please, checkout `feature/v4` branch of `https://github.com/ludeknovy/jtl-reporter` and just to be sure, delete the current one or check it out in a different folder. Then run `docker-compose up` again

**Comment by Vlad-Kliucharov** [comment]
@ludeknovy,  I had uploaded the "V4" branch and run the command "docker-compose up" again and on "Login" I hag an errors
![Screenshot_1](https://user-images.githubusercontent.com/20796574/140921410-88911023-5ab8-4153-afc7-d87a83a378f5.png)

**Reply by ludeknovy** [reply]
That's the same error as before :/. Did you really perform clean install as I recommended in my previous post (without any data left from the v3)?

**Reply by ludeknovy** [reply]
I just created a new VM and ran:
```
git clone https://github.com/ludeknovy/jtl-reporter.git
cd jtl-reporter
git checkout feature/v4

docker-compose up -d
```

and it worked like a charm for me.

**Reply by ludeknovy** [reply]
You can try it out here: http://161.35.218.96:2020/
user: vladyslav
password: jtl-reporter-{}

Will keep it up for a couple of hours.

**Reply by Vlad-Kliucharov** [reply]
But in this case above, it's working fine
![Screenshot_3](https://user-images.githubusercontent.com/20796574/140924287-4de0bd0a-cab4-4f09-b54b-00f1c9f6a563.png)

**Comment by Vlad-Kliucharov** [comment]
For me its not working "Locally", I had update all files

![Screenshot_1](https://user-images.githubusercontent.com/20796574/140924112-e7f477b5-0ee0-4525-855c-debba3a07a87.png)

**Reply by ludeknovy** [reply]
what's the content of your `db/Dockerfile` ?

**Reply by Vlad-Kliucharov** [reply]
Here it is
![Screenshot_4](https://user-images.githubusercontent.com/20796574/140926616-e0669a5f-4200-42b6-8be0-e5745accb9d1.png)

**Reply by ludeknovy** [reply]
Ok, that is the right one:). 
Now lets try to get the logs, please run `docker-compose logs be` (you must run in from the folder where you have your docker-compose file) and look for errors in the logs.

**Comment by Vlad-Kliucharov** [comment, marked as answer]
Now it's got the logs)

PS C:\Users\Archon> cd C:\jtl-reporter-feature-v4
PS C:\jtl-reporter-feature-v4> docker-compose logs be
jtl-reporter-feature-v4-be-1  |
jtl-reporter-feature-v4-be-1  | > jtl-reporter-be@1.0.0 start /src/be
jtl-reporter-feature-v4-be-1  | > node ./dist/server.js
jtl-reporter-feature-v4-be-1  |
jtl-reporter-feature-v4-be-1  | {"message":"Express server listening on port 5000","level":"info"}
jtl-reporter-feature-v4-be-1  | {"length":108,"name":"error","severity":"ERROR","code":"42P01","position":"15","file":"parse_relation.c","line":"1159","routine":"parserOpenTable","level":"error"}
jtl-reporter-feature-v4-be-1  | {"message":"Unexpected error: error: relation \"jtl.users\" does not exist","level":"error"}
jtl-reporter-feature-v4-be-1  | {"level":"info","message":"GET /api/info 500 53ms","meta":{}}
jtl-reporter-feature-v4-be-1  | {"length":108,"name":"error","severity":"ERROR","code":"42P01","position":"15","file":"parse_relation.c","line":"1159","routine":"parserOpenTable","level":"error"}
jtl-reporter-feature-v4-be-1  | {"message":"Unexpected error: error: relation \"jtl.users\" does not exist","level":"error"}
jtl-reporter-feature-v4-be-1  | {"level":"info","message":"GET /api/info 500 3ms","meta":{}}
jtl-reporter-feature-v4-be-1  | {"length":108,"name":"error","severity":"ERROR","code":"42P01","position":"15","file":"parse_relation.c","line":"1159","routine":"parserOpenTable","level":"error"}
jtl-reporter-feature-v4-be-1  | {"message":"Unexpected error: error: relation \"jtl.users\" does not exist","level":"error"}
jtl-reporter-feature-v4-be-1  | {"level":"info","message":"GET /api/info 500 19ms","meta":{}}

**Reply by ludeknovy** [reply]
ok, here we go: ` {"message":"Unexpected error: error: relation "jtl.users" does not exist","level":"error"}`

So that means the initial DB setup did not run correctly. What's your output of `docker-compose logs db`

**Reply by Vlad-Kliucharov** [reply]
I don't understand? I had sent logs above

**Reply by ludeknovy** [reply]
Sorry, I meant `db` instead of `be`. Updated the command above.

**Reply by Vlad-Kliucharov** [reply]
PS C:\jtl-reporter-feature-v4> docker-compose logs db
jtl-reporter-feature-v4-db-1  | ********************************************************************************
jtl-reporter-feature-v4-db-1  | WARNING: POSTGRES_HOST_AUTH_METHOD has been set to "trust". This will allow
jtl-reporter-feature-v4-db-1  |          anyone with access to the Postgres port to access your database without
jtl-reporter-feature-v4-db-1  |          a password, even if POSTGRES_PASSWORD is set. See PostgreSQL
jtl-reporter-feature-v4-db-1  |          documentation about "trust":
jtl-reporter-feature-v4-db-1  |          https://www.postgresql.org/docs/current/auth-trust.html
jtl-reporter-feature-v4-db-1  |          In Docker's default configuration, this is effectively any other
jtl-reporter-feature-v4-db-1  |          container on the same system.
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  |          It is not recommended to use POSTGRES_HOST_AUTH_METHOD=trust. Replace
jtl-reporter-feature-v4-db-1  |          it with "-e POSTGRES_PASSWORD=password" instead to set a password in
jtl-reporter-feature-v4-db-1  |          "docker run".
jtl-reporter-feature-v4-db-1  | ********************************************************************************
jtl-reporter-feature-v4-db-1  | The files belonging to this database system will be owned by user "postgres".
jtl-reporter-feature-v4-db-1  | This user must also own the server process.
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | The database cluster will be initialized with locale "en_US.utf8".
jtl-reporter-feature-v4-db-1  | The default database encoding has accordingly been set to "UTF8".
jtl-reporter-feature-v4-db-1  | The default text search configuration will be set to "english".
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | Data page checksums are disabled.
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | fixing permissions on existing directory /var/lib/postgresql/data ... ok
jtl-reporter-feature-v4-db-1  | creating subdirectories ... ok
jtl-reporter-feature-v4-db-1  | selecting default max_connections ... 100
jtl-reporter-feature-v4-db-1  | selecting default shared_buffers ... 128MB
jtl-reporter-feature-v4-db-1  | selecting default timezone ... Etc/UTC
jtl-reporter-feature-v4-db-1  | selecting dynamic shared memory implementation ... posix
jtl-reporter-feature-v4-db-1  | creating configuration files ... ok
jtl-reporter-feature-v4-db-1  | running bootstrap script ... ok
jtl-reporter-feature-v4-db-1  | performing post-bootstrap initialization ... ok
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | WARNING: enabling "trust" authentication for local connections
jtl-reporter-feature-v4-db-1  | You can change this by editing pg_hba.conf or using the option -A, or
jtl-reporter-feature-v4-db-1  | --auth-local and --auth-host, the next time you run initdb.
jtl-reporter-feature-v4-db-1  | syncing data to disk ... ok
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | Success. You can now start the database server using:
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  |     pg_ctl -D /var/lib/postgresql/data -l logfile start
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | waiting for server to start....2021-11-09 12:20:14.619 UTC [51] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:14.650 UTC [52] LOG:  database system was shut down at 2021-11-09 12:20:11 UTC
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:14.673 UTC [51] LOG:  database system is ready to accept connections
jtl-reporter-feature-v4-db-1  |  done
jtl-reporter-feature-v4-db-1  | server started
jtl-reporter-feature-v4-db-1  | CREATE DATABASE
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/schema.sql
jtl-reporter-feature-v4-db-1  | CREATE EXTENSION
jtl-reporter-feature-v4-db-1  | CREATE SCHEMA
jtl-reporter-feature-v4-db-1  | CREATE TABLE
jtl-reporter-feature-v4-db-1  | CREATE TABLE
jtl-reporter-feature-v4-db-1  | CREATE TABLE
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:16.798 UTC [51] LOG:  received fast shutdown request
jtl-reporter-feature-v4-db-1  | waiting for server to shut down...2021-11-09 12:20:16.800 UTC [51] LOG:  aborting any active transactions
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:16.801 UTC [51] LOG:  background worker "logical replication launcher" (PID 58) exited with exit code 1
jtl-reporter-feature-v4-db-1  | .2021-11-09 12:20:16.815 UTC [53] LOG:  shutting down
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:16.977 UTC [51] LOG:  database system is shut down
jtl-reporter-feature-v4-db-1  |  done
jtl-reporter-feature-v4-db-1  | server stopped
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | PostgreSQL init process complete; ready for start up.
jtl-reporter-feature-v4-db-1  |
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:17.098 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:17.098 UTC [1] LOG:  listening on IPv6 address "::", port 5432
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:17.101 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:17.163 UTC [97] LOG:  database system was shut down at 2021-11-09 12:20:16 UTC
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:17.186 UTC [1] LOG:  database system is ready to accept connections
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:24.272 UTC [104] ERROR:  relation "jtl.users" does not exist at character 15
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:24.272 UTC [104] STATEMENT:  SELECT * FROM jtl.users
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:30.776 UTC [104] ERROR:  relation "jtl.users" does not exist at character 15
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:20:30.776 UTC [104] STATEMENT:  SELECT * FROM jtl.users
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:43:44.526 UTC [1311] ERROR:  relation "jtl.users" does not exist at character 15
jtl-reporter-feature-v4-db-1  | 2021-11-09 12:43:44.526 UTC [1311] STATEMENT:  SELECT * FROM jtl.users

**Reply by ludeknovy** [reply]
Thanks. So this means you still run the DB for v3.
Try:
1.  `docker-compose down`
2. remove `data` folder 
3. `docker-compose up -d --build --force-recreate`

**Reply by Vlad-Kliucharov** [reply]
@ludeknovy, perfect, now it's working!

**Reply by ludeknovy** [reply]
What a relief 😆 . Thanks for your patience.

**Reply by Vlad-Kliucharov** [reply]
Thanks for your help!
